ssl/quic/quic_lcidm.c

```diff
@@ -554,3 +600,17 @@ int ossl_quic_lcidm_debug_add(QUIC_LCIDM *lcidm, void *opaque,
     lcid_obj->type      = LCID_TYPE_NCID;
     return 1;
 }
+
+int ossl_quic_lcidm_get_unused_cid(QUIC_LCIDM *lcidm, QUIC_CONN_ID *cid)
+{
+    int i;
+
+    for (i = 0; i < 10; i++) {
+        if (lcidm_generate_cid(lcidm, cid))
```

Comment:
All this does is generate a random cid. It doesn't add it to the lcidm. Don't you need to call `lcidm_conn_new_lcid` to actually add it? Compare with `lcidm_generate`

Reply:
my intention is:
   - generate random lcid which is not used (happens in snippet above)
   - use the random lcid as source con id in retry packet
   - the next initial packet which carries validation token uses lcid as destination connection id
   - the `port_bind_channel()` function creates channel and inserts `lcid` (`dstConnId` from packet header) to `lcidm`
 
the bug must be hiding in the codepath which implements the last step above.